carservice/src/test/java/com/senlainc/warsaw/tyurin/GaragePlaceServiceUnitTest.java

```diff
@@ -0,0 +1,117 @@
+package com.senlainc.warsaw.tyurin;
+
+import com.senlainc.warsaw.tyurin.dao.IGaragePlaceDao;
+import com.senlainc.warsaw.tyurin.entity.GaragePlace;
+import com.senlainc.warsaw.tyurin.exception.NotFoundException;
+import com.senlainc.warsaw.tyurin.service.GaragePlaceService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class GaragePlaceServiceUnitTest {
+
+    @InjectMocks
+    private GaragePlaceService garagePlaceService;
+    @Mock
+    private IGaragePlaceDao garagePlaceDao;
+
+    @AfterEach
+    void tearDown() {
+        Mockito.verifyNoMoreInteractions(garagePlaceDao);
+    }
+
+    @Test
+    void delete_deleteGaragePlaceById_deletedSuccessfully() throws NotFoundException {
+        GaragePlace testGaragePlace = new GaragePlace();
+        testGaragePlace.setId(1L);
+        testGaragePlace.setNumber(1);
+        testGaragePlace.setSpace(15.0);
+
+        ReflectionTestUtils.setField(garagePlaceService,"isGaragePlaceRemovable",true);
```

Comment:
Using ReflectionUtils in tests is generally discouraged due to several reasons. Reflection in Java provides dynamic access to fields, methods, and constructors during runtime, but it comes with trade-offs that can impact code readability, maintainability, and robustness. Here are some key reasons why it's not recommended:
   - Code that relies heavily on reflection can be difficult to read and understand, making it harder for developers to grasp the logic and intentions behind the tests.
   - Reflection bypasses the usual compile-time checks, making the code more prone to errors. If the structure of the classes being reflected upon changes, it can break the tests, and these issues might not be caught until runtime.
   - Many testing tools and frameworks are designed to work seamlessly with standard testing practices. Reflection-based code may not be as compatible with these tools, leading to a lack of support for automated testing and other testing utilities.
   - Debugging reflective code can be more challenging than debugging regular code. Developers may find it harder to trace and understand the flow of the program during debugging sessions.
   - Reflection can have performance overhead, and using it excessively in tests may impact the overall test suite performance. Tests should ideally run quickly and efficiently.
   - The need for extensive use of reflection in tests might indicate design issues in the codebase. It's often better to address these design concerns rather than resorting to reflection for testing purposes.

Reply:
To fix issue with value you can use test resources, with desired data

@TestPropertySource(locations = "classpath:test-application.properties")

Reply:
Thank you for detailed explanation
@TestPropertySource(locations = "classpath:test-application.properties") is what I need in this situation